frontend/helpers/tillDuration.ts

```diff
@@ -25,3 +26,17 @@ export const formatInterval = function (startDate: Date, endDate: Date): string
         });
     }
 };
+
+export const timeTillProfitable = function (auction: AuctionTransaction) {
+    let steps = 0;
+    let currentPrice = auction.approximateUnitPrice.toNumber();
+
+    const currentStepProgress = auction.secondsBetweenPriceDrops!! - auction.secondsTillNextPriceDrop!!;
+
+    while (currentPrice > auction.marketUnitPrice!!.toNumber()) {
```

Comment:
I was just going through your PR and saw that you were calculating the price with a loop. I remember this was also how I initially approached it, until I figured out it could be done in a more effective way with math. Maybe have a look at this file to see how I calculated prices (you could also reuse some of the functions there or maybe move this entire function into that file as it relates to it).

https://github.com/sidestream-tech/unified-auctions-ui/blob/main/core/src/price.ts